java-checks/src/main/java/org/sonar/java/checks/AvoidStarImportCheck.java

```diff
@@ -0,0 +1,119 @@
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.ImportTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+/**
+ *
+ * Squid implementation for checkstyle <a href="checkstyle.sourceforge.net/config_imports.html#AvoidStarImport">AvoidStarImport</a> rule
+ *
+ */
+@Rule(
+  key = "AvoidStarImportCheck",
+  name = "Checks that there are no import statements that use the * notation",
+  tags = {"convention", "bug"},
+  priority = Priority.MINOR)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
+@SqaleConstantRemediation("1min")
+public class AvoidStarImportCheck extends IssuableSubscriptionVisitor {
+
+  private static final String DEFAULT_EXCLUDE = "";
+
+  private static final boolean DEFAULT_ALLOW = false;
+
+  @RuleProperty(
+    key = "AllowClassImports",
+    description = "Whether to allow starred class imports like import java.util.*;",
+    defaultValue = "" + DEFAULT_ALLOW)
+  public boolean allowClassImports = DEFAULT_ALLOW;
+
+  @RuleProperty(
+    key = "AllowStaticMemberImports",
+    description = "whether to allow starred static member imports like import static org.junit.Assert.*",
+    defaultValue = "" + DEFAULT_ALLOW)
+  public boolean allowStaticMemberImports = DEFAULT_ALLOW;
+
+  @RuleProperty(
+    key = "Exclude",
+    description = "Comma separated list of packages where star imports are allowed (Note that this property is not recursive, subpackages of excluded packages are not automatically excluded)",
+    defaultValue = "" + DEFAULT_EXCLUDE)
+  public String exclude = DEFAULT_EXCLUDE;
+  private HashSet<String> excludePackages = new HashSet<>();
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    if (StringUtils.isNotBlank(exclude)) {
+      // Add .* suffix to accelerate the match
+      for (String p : exclude.split(",")) {
+        excludePackages.add(p + ".*");
+      }
+    }
+    return ImmutableList.of(Tree.Kind.IMPORT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    // Visit import only => cast
+    ImportTree importTree = (ImportTree) tree;
+    String importName = fullQualifiedName(importTree.qualifiedIdentifier());
+
+    // There is a star in the import
+    if (importName.endsWith(".*")) {
+      boolean allowed = allowClassImports && !importTree.isStatic();
+      boolean allowedStatic = allowStaticMemberImports && importTree.isStatic();
+      boolean excluded = !excludePackages.isEmpty() && excludePackages.contains(importName);
+      if (!allowed && !allowedStatic && !excluded) {
+        addIssue(importTree, String.format("Using the '.*' form of import should be avoided - %s", importName));

```

Comment:
as message should comply with http://jira.sonarsource.com/browse/RSPEC-2208 and with removal of package exclusion property this should allow to drop computation of name, simplifying the check.